tests/nativeapp/test_run_processor.py

```diff
@@ -1300,6 +1296,63 @@ def test_upgrade_app_fails_upgrade_restriction_error(
     assert mock_execute.mock_calls == expected
 
 
+@mock.patch(NATIVEAPP_MANAGER_EXECUTE)
+@mock.patch(RUN_PROCESSOR_GET_EXISTING_APP_INFO)
+@mock_connection()
+def test_versioned_app_x_upgrade_to_unversioned(
+    mock_conn,
+    mock_get_existing_app_info,
+    mock_execute,
+    temp_dir,
+    mock_cursor,
+):
+    mock_get_existing_app_info.return_value = {
+        "name": "myapp",
+        "comment": SPECIAL_COMMENT,
+        "owner": "app_role",
+        "version": "v1",
+    }
+    side_effects, expected = mock_execute_helper(
+        [
+            (
+                mock_cursor([{"CURRENT_ROLE()": "old_role"}], []),
+                mock.call("select current_role()", cursor_class=DictCursor),
+            ),
+            (None, mock.call("use role app_role")),
+            (None, mock.call("use warehouse app_warehouse")),
+            (
+                ProgrammingError(
+                    msg="Some Error Message.",
+                    errno=93045,
+                ),
+                mock.call(
+                    "alter application myapp upgrade using @app_pkg.app_src.stage"
+                ),
+            ),
+            (None, mock.call("use role old_role")),
+        ]
+    )
+    mock_conn.return_value = MockConnectionCtx()
+    mock_execute.side_effect = side_effects
+
+    current_working_directory = os.getcwd()
+    create_named_file(
+        file_name="snowflake.yml",
+        dir_name=current_working_directory,
+        contents=[mock_snowflake_yml_file],
+    )
+
+    run_processor = _get_na_run_processor()
+    with pytest.raises(typer.Exit):
+        result = run_processor.create_or_upgrade_app(
+            policy=DenyAlwaysPolicy(),
+            is_interactive=False,
+            install_method=SameAccountInstallMethod.unversioned_dev(),
+        )
+        assert result.exit_code == 1
```

Comment:
I am having hard time understanding what this test is doing. Shouldn't we expect this to succeed since we now allow versioned to unversioned upgrade? If so, why are we checking for exit_code 1?

Reply:
Looks this is set up as a negative test (see the fixture creation), but like Michel, I wonder about the absence of a positive test here.

Reply:
Yeah, this is to make sure transitioning to an unversioned app goes through the drop-on-upgrade flow which raises this typer exit. A positive test makes sense; I'll add an integration one.